workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewFactory.kt

```diff
@@ -93,3 +283,201 @@ public fun interface ViewStarter {
+ *      }
+ *
+ * To make one rendering type a wrapper for others:
+ *
+ *    class Wrapper<W>(val wrapped: W: Screen) : Screen, Compatible {
+ *      override val compatibilityKey = Compatible.keyFor(wrapped)
+ *    }
+ *
+ *    fun <W : Screen> WrapperViewFactory() =
+ *      ScreenViewFactory.forBuiltView<Wrapper<W>> { wrapper, env, context, container ->
+ *        // Get the view factory of the wrapped screen.
+ *        wrapper.wrapped.toViewFactory(env)
+ *          // Transform it to factory that accepts Wrapper<W>
+ *          .unwrapping<Wrapper<W>, W> { it.wrapped }
+ *          // Delegate to the transformed factory to build the view.
+ *          .buildView(wrapper, env, context, container)
+ *      }
+ *
+ * To make a wrapper that adds information to the [ViewEnvironment]:
+ *
+ *    class NeutronFlowPolarity(val reversed: Boolean) {
+ *      companion object : ViewEnvironmentKey<NeutronFlowPolarity>(
+ *        NeutronFlowPolarity::class
+ *      ) {
+ *        override val default: NeutronFlowPolarity =
+ *          NeutronFlowPolarity(reversed = false)
+ *      }
+ *    }
+ *
+ *    class OverrideNeutronFlow<W : Screen>(
+ *      val wrapped: W,
+ *      val polarity: NeutronFlowPolarity
+ *    ) : Screen, Compatible {
+ *      override val compatibilityKey: String = Compatible.keyFor(wrapped)
+ *    }
+ *
+ *    fun <W : Screen> OverrideNeutronFlowViewFactory() =
+ *      ScreenViewFactory.forBuiltView<OverrideNeutronFlow<W>> { wrapper, env, context, container ->
+ *        // Get the view factory of the wrapped screen.
+ *        wrapper.wrapped.toViewFactory(env)
+ *          // Transform it to factory that accepts OverrideNeutronFlow<W>, by
+ *          // replacing the OverrideNeutronFlow<W> with an EnvironmentScreen<W>
+ *          .unwrapping<OverrideNeutronFlow<W>, EnvironmentScreen<W>> {
+ *            it.wrapped.withEnvironment(
+ *              Environment.EMPTY + (NeutronFlowPolarity to it.polarity)
+ *            )
+ *          }
+ *          // Delegate to the transformed factory to build the view.
+ *          .buildView(wrapper, env, context, container)
+ *      }
+ *
+ * @param unwrap a function to extract [WrappedT] instances from [WrapperT]s.
+ */
+@WorkflowUiExperimentalApi
+public inline
+fun <reified WrapperT : Screen, WrappedT : Screen> ScreenViewFactory<WrappedT>.unwrapping(
+  crossinline unwrap: (wrapperScreen: WrapperT) -> WrappedT,
+): ScreenViewFactory<WrapperT> = unwrapping(unwrap) { _, ws, e, su -> su(unwrap(ws), e) }
+
+/**
+ * Transforms a [ScreenViewFactory] of [WrappedT] into one that can handle
+ * instances of [WrapperT].
+ *
+ * One general note: when creating a wrapper rendering, you're very likely to want it
+ * to implement [Compatible], to ensure that checks made to update or replace a view
+ * are based on the wrapped item. Each wrapper example below illustrates this.
+ *
+ * Also see the simpler variant of this function that takes only an [unwrap] argument.
+ *
+ * ## Example
```

Comment:
It sounded to me like the two examples were different in shape, but they look the same. Same structure, same calls? (not saying it's bad, just asking if I'm missing any difference)

Reply:
You know, you're right. I originally had dropped the second one, but then tired me thought it actually covered more ground.

It really doesn't, deleting it.